docs/pages/guides/server-components.mdx

```diff
@@ -388,37 +327,112 @@ async function ExpensiveTask() {
 
-```tsx app/index.tsx
-import { Text, View } from 'react-native';
+```tsx actions/renderData.tsx
+// This will assert if the module runs on the client.
+import 'server-only';
 
-export const unstable_settings = {
-  // This component will be rendered at build-time and never re-rendered in production.
-  render: 'static',
-};
+import { Text } from 'react-native';
+
+export async function renderData() {
+  // This code only runs on the server.
+  const data = await fetch('https://my-endpoint/', {
+    headers: {
+      Authorization: `Bearer ${process.env.SECRET}`,
+    },
+  });
+
+  // ...
+  return <div />;
+}
+```
+
+You can define the secret in your **.env** file:
+
+```text .env
+SECRET=123
+```
+
+> You do not need to restart the dev server to update environment variables. They are automatically reloaded on every request.
+
+## Platform detection
+
+To detect which platform your code is bundled for, use the `process.env.EXPO_OS` environment variable. For example, `process.env.EXPO_OS === 'ios'`. Prefer this to `Platform.OS` as `react-native` is not fully optimized for React Server Components yet and won't work as expected.
 
+You can detect if code is running in the server by performing a `typeof window === 'undefined'` check. This will always return `true` on client devices and `false` on the server.
+
+## Testing with jest
+
+Library authors can test their modules support Server Components by using `jest-expo`. Learn more in the [testing react server components guide](/guides/testing-rsc).
+
+## Metadata
+
+React Server Components are a feature of React 19. To enable them, Expo CLI automatically uses a special canary build of React on all platforms. In the future, it will be removed when React 19 is enabled by default in React Native.
+
+As a result, you can use React 19 features such as placing `<meta>` tags anywhere in your app (web-only).
+
+```tsx app/index.tsx
 export default function Index() {
   return (
-    <View>
-      <Text>Hello, world!</Text>
-    </View>
+    <>
+      {process.env.EXPO_OS === 'web' && (
+        <>
+          <meta name="description" content="Hello, world!" />
+          <meta property="og:image" content="/og-image.png" />
+        </>
+      )}
+      <MyComponent />
+    </>
   );
 }
 ```
 
-- `render: 'static'` will render the component at build-time and never re-render it in production. This is similar to how classic static site generators work.
-- `render: 'dynamic'` will render the component at request-time and re-render it on every request. This is similar to how server-side rendering works.
+You can use this instead of the `Head` component from `expo-router/head`, but it only works on web for now.
 
-If you want client-side rendering, move your data fetching to a Client Component and control the rendering locally.
+## Known limitations
```

Comment:
It'll be ideal to move this limitation section at the end of the document. It's a good pattern in general when scanning a document from a reader's eye and something we also follow in other documents.